backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts

```diff
@@ -905,42 +928,19 @@ export class ExperimentAssignmentService {
 
     if (groupCount && userCount && experiment.assignmentUnit === ASSIGNMENT_UNIT.GROUP) {
       // fetch all the monitored document if exist
-      const monitoredDocuments = await getMonitoredDocumentOfExperiment(experiment);
-
-      // check for student inside each group
-      const groupMap = new Map<string, Set<string>>();
-      // groupAssignments.forEach((groupAssignment) => {
-      //   groupMap.set(groupAssignment.groupId, []);
-      // });
-      monitoredDocuments.forEach((monitoredDocument) => {
-        const groupId = monitoredDocument.user.workingGroup[group];
-        // create new Set for new group
-        if (!groupMap.has(groupId)) {
-          groupMap.set(groupId, new Set());
-        }
-        groupMap.set(groupId, groupMap.get(groupId).add(monitoredDocument.user.id));
-      });
-      let groupSatisfied = 0;
-
-      groupMap.forEach((groupElement) => {
-        if (groupElement.size >= userCount) {
-          groupSatisfied++;
-        }
-      });
+      const monitoredDocuments = await getMonitoredDocumentOfExperimentTypeGroup(experiment);
 
-      if (groupSatisfied >= groupCount) {
+      // check if the group count is more than ending criteria
+      if (Object.keys(monitoredDocuments).length >= groupCount) {
```

Comment:
Minor thing, but if monitoredDocuments is an array `Object.keys` is not necessary, `monitoredDocuments.length` would give same count.

Reply:
ok, thank you, will update the code.